query/src/main/java/org/infinispan/query/dsl/embedded/impl/EmbeddedLuceneQuery.java

```diff
@@ -115,6 +115,39 @@ public String toString() {
             '}';
    }
 
+   private IndexedQuery<T> getOrCreateIndexedQuery(boolean withHitCount) {
+      if (withHitCount) {
+         if (indexedQuery == null) {
+            indexedQuery = createQuery(true);
+         }
+         return indexedQuery;
+      } else {
+         if (indexedListQuery == null) {
+            indexedListQuery = createQuery(false);
+         }
+         return indexedListQuery;
+      }
+   }
+
+   private IndexedQuery<T> createQuery(boolean withHitCount) {
+      validateNamedParameters();
+      IndexedQuery<T> result = queryEngine.buildLuceneQuery(parsingResult, namedParameters, startOffset, maxResults, isLocal());
+
+      if (withHitCount) {
+         if (hitCountAccuracy.isPresent()) {
+            result = result.hitCountAccuracy(hitCountAccuracy.get());
+         }
+      } else {
+         result.hitCountAccuracy(1); // lower the hit count accuracy
```

Comment:
Shouldn't this value be the same as the "default" value that we have in the configuration? From what I saw the default was 10,000.

Reply:
Before this contribution the default (and fixed value for it) was Integer.MAX :) I think that 10,000 is a good trade off. That is the same default applied by the Elastic https://www.elastic.co/guide/en/elasticsearch/reference/master/search-your-data.html#track-total-hits.
This contribution is likely to improve the performance of all the query with some conditions on indexes that have more than 10,000 elements... that's the reason I would like to have it soon in our code base.

Reply:
To address your comment... In this case we don't need any hit count so we can lower the hit count accuracy to 1.